### PR TITLE
Add document for testing component accessibility

### PR DESCRIPTION
**Pull request by EoinShaughnessy**
Co-authored by: @hannalaakso 

## What we're adding

This PR fixes [#1781](https://github.com/alphagov/govuk-design-system/issues/1781) and adds a new document ('Test components using accessibility acceptance criteria') to the ['Contributing' folder in our Frontend repo](https://github.com/alphagov/govuk-frontend/tree/main/docs/contributing).

## Why we're adding it

When building the notification and cookie banners, we used a template to check how components should behave with assistive tech. We feel there is value in publishing this information so contributors can use it, too.

### CODE REVIEW

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (vanitabarrett) [inline docs/contributing/test-components-using-accessibility-acceptance-criteria.md:12]: This link needs to be updated

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (vanitabarrett) [inline docs/contributing/test-components-using-accessibility-acceptance-criteria.md:13]: Looks like this is intended to be a link too? Link text needs defining and the link URL updating

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (vanitabarrett) [inline docs/contributing/test-components-using-accessibility-acceptance-criteria.md:58]: Just wanted to check - are we intentionally linking to https://dequeuniversity.com/screenreaders/ twice?

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (vanitabarrett) [inline docs/contributing/test-components-using-accessibility-acceptance-criteria.md:57]:
```suggestion
- a screen reader in browse or reader mode
```

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (vanitabarrett) [inline docs/contributing/test-components-using-accessibility-acceptance-criteria.md:66]: This page seems to be talking specifically about the `main` landmark, but there are other landmark regions. Can we link anywhere more generic? Maybe https://www.w3.org/TR/wai-aria-practices/examples/landmarks/index.html ?

**Review by vanitabarrett — CHANGES_REQUESTED** [review]
Looks great @EoinShaughnessy ! Just a couple of small things I noticed that need fixing

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (EoinShaughnessy) [reply docs/contributing/test-components-using-accessibility-acceptance-criteria.md:58]: Yeah, that's intentional. Bit clunky, I know. Our rationale was, including it twice hopefully makes it less likely users will pass over accessing content that could help them.

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (EoinShaughnessy) [reply docs/contributing/test-components-using-accessibility-acceptance-criteria.md:66]:
```suggestion
- be labelled as a [landmark region](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/index.html), if appropriate
```

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (EoinShaughnessy) [reply docs/contributing/test-components-using-accessibility-acceptance-criteria.md:12]: My bad! Thanks for flagging.
```suggestion
- [the GOV.UK Design System’s accessibility acceptance criteria](https://github.com/alphagov/govuk-frontend/blob/071ad5032e1d94f6fbcf34fd248c062c923b792d/docs/contributing/test-components-using-accessibility-acceptance-criteria.md#govuk-design-system---accessibility-acceptance-criteria)
```

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (EoinShaughnessy) [reply docs/contributing/test-components-using-accessibility-acceptance-criteria.md:13]:
```suggestion
- custom accessibility acceptance criteria that you need to write specifically for your component or iteration(https://github.com/alphagov/govuk-frontend/blob/071ad5032e1d94f6fbcf34fd248c062c923b792d/docs/contributing/test-components-using-accessibility-acceptance-criteria.md#writing-custom-accessibility-acceptance-criteria-for-your-component
```

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (vanitabarrett) [inline docs/contributing/test-components-using-accessibility-acceptance-criteria.md:12]: @EoinShaughnessy Because this is a link to a heading within the same document, we should be able to do something like this:

```suggestion
- [the GOV.UK Design System’s accessibility acceptance criteria](#govuk-design-system---accessibility-acceptance-criteria)
```

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (vanitabarrett) [inline docs/contributing/test-components-using-accessibility-acceptance-criteria.md:13]: @EoinShaughnessy Same comment as above, as this is a link to a heading in the same document we can use an [anchor link](https://gist.github.com/asabaylus/3071099), i.e: you can just have everything after the 'hashtag' #, you don't need the full URL.

You're also missing the markdown link syntax here, e.g: [custom accessibility acceptance...etc](link path)

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (EoinShaughnessy) [reply docs/contributing/test-components-using-accessibility-acceptance-criteria.md:12]: Ah great, thanks @vanitabarrett

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (EoinShaughnessy) [reply docs/contributing/test-components-using-accessibility-acceptance-criteria.md:13]:
```suggestion
- custom accessibility acceptance criteria that you need to write specifically for your component or iteration(#writing-custom-accessibility-acceptance-criteria-for-your-component
```

File: docs/contributing/test-components-using-accessibility-acceptance-criteria.md
Comment (EoinShaughnessy) [inline docs/contributing/test-components-using-accessibility-acceptance-criteria.md:13]:
```suggestion
- [custom accessibility acceptance criteria that you need to write specifically for your component or iteration](#writing-custom-accessibility-acceptance-criteria-for-your-component)
```

**Comment by vanitabarrett** [comment]
@EoinShaughnessy Looks good! Are you ok to squash the commits before merging?

**Comment by EoinShaughnessy** [comment]
@vanitabarrett Cheers! Yeah, I'll sort the commits (but thanks, I appreciate the offer). :)

**Review by vanitabarrett — APPROVED** [review]

**Comment by EoinShaughnessy** [comment]
Holding off merging till we can attribute proper co-authorship!

**Pull request merged**
